### DISCUSSION TITLE
ImageMagick Edge case handling

### DISCUSSION THREAD

**Discussion by awesomecoder111** [ImageMagick/ImageMagick#4420] [Development]
It appears that ImageMagick handles edge cases regarding image inputs when using the convert command to generate thumbnails better than other image processing libraries. Three cases i've found so far are
1. if image has corrupted end of file character, it will still work
2. if the image has an orientation specified in the exif metadata, it will automatically orient it based on that value
3. if image provided is a gzip file, image magick will uncompress it before

is there any documentation explaining what edge cases are handled or places in the code where I can look to see which other edge cases are covered?
I'm also confused how the image is automatically being oriented correctly even when I don't pass this [flag](https://github.com/ImageMagick/ImageMagick/blob/2747ccfc1044fc3da6d32ff1ebbca5e926fcf602/MagickWand/convert.c#L159), for another library I have to make sure to pass a similar flag to make it work

**Comment by fmw42** [comment]
The flag is in the EXIF data in the header.  What do you mean by "edge"? I do not see any connection to edges in the image nor the bounding sides of the image. Please clarify "edge" in your usage.

**Reply by awesomecoder111** [reply]
by edge cases i mean invalid/strange input images such as missing end of file character or gzip image file.

e.g. while using sharp library i had to disable `failOnError` flag to get it to work, but image magick just handles it https://github.com/lovell/sharp/issues/1859

**Reply by awesomecoder111** [reply]
but i thought the `-auto-orient` flag would be needed in order for it to use exif data, but it orients without passing this flag

**Reply by fmw42** [reply]
Please provide an example image that works without the -auto-orient flag in the exif data and is not the default orientation.

**Comment by snibgo** [comment]
What do you mean by "edge cases"?

> 1. if image has corrupted end of file character, it will still work

Please provide an example. If IM tries to read a file that does not conform to standards, it should report an error.

"-auto-orient" is documented at https://imagemagick.org/script/command-line-options.php#auto-orient

**Reply by awesomecoder111** [reply]
https://github.com/ImageMagick/ImageMagick/discussions/4420#discussioncomment-1556316

For some reason the imageMagick handles the exif rotation case automatically while for another library i had to add the auto-orient flag.

**Reply by fmw42** [reply]
You should always use -auto-orient to ensure it gets properly rotated. It should not work without it unless the image has the proper default orientation already.

However, there have been some cases reported where the flag was set wrong.
